src/models/player.history.spec.js

```diff
@@ -0,0 +1,164 @@
+      beforeEach(function () {
+        this.result = PlayerHistory.manyFromDbEntry([]);
+      });
+
+      it("the returned array's will be empty", function () {
+        expect(this.result).toEqual([]);
+      });
+    });
+  });
+
+  describe(".fromRawData instantiates a new PlayerHistory object and returns it.", function () {
+    describe("When players and a date are passed in,", function () {
+      beforeEach(function () {
+        this.players = 62;
+        this.date = "September 2014";
+
+        this.result = PlayerHistory.fromRawData(this.players, this.date);
+      });
+
+      it("the result is an instance of PlayerHistory.", function () {
+        expect(this.result).toBeInstanceOf(PlayerHistory);
+      });
+      it("the result has a property called year, which equals 2014.", function () {
+        expect(this.result.year).toBe(2014);
+      });
+      it("the result has a property called month, which equals 8.", function () {
+        expect(this.result.month).toBe(8);
+      });
+      it("the result has a property called averagePlayers, which equals 62", function () {
+        expect(this.result.averagePlayers).toBe(62);
+      });
+      it("the result has a property called tracked players, which is an instance of TrackedPlayers", function () {
+        expect(this.result.trackedPlayers[0]).toBeInstanceOf(TrackedPlayers);
+      });
+      it("the result has a property called players, which equals 62", function () {
+        expect(this.result.trackedPlayers[0].players).toBe(62);
+      });
+    });
+  });
+
+  describe(".newMonthlyEntry instantiates the PlayerHistory class with default values. The returned object", function () {
+    beforeEach(function () {
+      this.currentDate = new Date();
+
+      this.result = PlayerHistory.newMonthlyEntry(42);
+      debugger;
+    });
+
+    it("is an instance of PlayerHistory", function () {
+      expect(this.result).toBeInstanceOf(PlayerHistory);
+    });
+    it("has a property called 'year'. It's value equals the current year", function () {
+      expect(this.result.year).toBe(this.currentDate.getFullYear());
+    });
+    it("has a property called 'month'. It's value equals the current month", function () {
+      expect(this.result.month).toBe(this.currentDate.getMonth());
+    });
+    it("has a property called 'averagePlayers'. It's value equals '42'", function () {
+      expect(this.result.averagePlayers).toBe(42);
+    });
+    it("has a property called 'trackedPlayers'. It is an instance of TrackedPlayers", function () {
+      expect(this.result.trackedPlayers[0]).toBeInstanceOf(TrackedPlayers);
+    });
+    it("has a property called 'players'. It's value equals '42'", function () {
+      expect(this.result.trackedPlayers[0].players).toBe(42);
+    });
+  });
+
+  describe(".pushTrackedPlayers adds an new instance of TrackedPlayers, and updates the average players property. The modified object", function () {
+    beforeEach(function () {
```

Comment:
I thought adding different months and years and the same `PlayerHistory` instance is not possible? Isn't the idea to have in one `PlayerHistory` instance many TrackedPlayer entries for **one month** in **one year**? If there is a new month and a new year, there should be a new `PlayerHistory` instance, or? Please let me know and update the tests if I am right.

For the `it` blocks I propose the same style of description and checks as described above.